File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    stack = []
    cnt = 0
    for food in ingredient:
        stack.append(food)
        # 스택의 마지막 4개 요소가 "1231"인지 확인
        if len(stack) >= 4 and "".join(map(str, stack[-4:])) == "1231":
            cnt += 1
            stack.pop()
            stack.pop()
            stack.pop()
            stack.pop()
    return cnt

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    const hamburger = [1, 2, 3];
    
    let stack = [];
    let cnt = 0;
    for(const food of ingredient){
        stack.push(food);
        if(stack.length >= 4 && stack.slice(-4).join("") === "1231"){
            cnt++;
            stack.splice(-4);
        }
    }
    return cnt;
}